marlowe-playground-client/src/Marlowe/Holes.purs

```diff
@@ -205,13 +213,26 @@ constructMarloweType constructorName (MarloweHole { row, column }) m = case Map.
 
   showArgument _ NewtypeArg = ""
 
+  showArgument _ (GenArg marloweType) = case getConstructorFromMarloweType marloweType of
+    Nothing -> "?hole"
+    Just name ->
+      let
+        newArgs = getMarloweConstructors marloweType
+
+        hole = MarloweHole { name, marloweType, row: 0, column: 0 }
+      in
+        constructMarloweType name hole newArgs
+
   parens 1 1 text = text
 
   parens _ _ text = "(" <> text <> ")"
 
 mkHole :: forall a. String -> { row :: Pos, column :: Pos } -> Term a
 mkHole name position = Hole name Proxy position
 
+mkDefaultTerm :: forall a. a -> Term a
+mkDefaultTerm a = Term a { row: 0, column: 0 }
+
 data Term a
   = Term a { row :: Pos, column :: Pos }
   | Hole String (Proxy a) { row :: Pos, column :: Pos }
```

Comment:
This raises a (wholly academic) question, could you give `Term` a `Functor` instance? You'd need a functor instance for `Proxy`, but it's there, so...yes.

Interesting.

Reply:
an Applicative might make sense even. But... not right now